akka-javasdk/src/main/java/akka/javasdk/annotations/Agent.java

```diff
@@ -0,0 +1,25 @@
+/*
+ * Copyright (C) 2021-2024 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.javasdk.annotations;
+
+import java.lang.annotation.Documented;
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+/**
+ * Assign an identifier and description to a ChatAgent component.
+ * <p>
+ * The identifier should be unique among the different components.
+ */
+@Target(ElementType.TYPE)
+@Retention(RetentionPolicy.RUNTIME)
+@Documented
+public @interface Agent {
+  String value();
```

Comment:
This is the `id`, maybe it should be named `id`? `value` has some special meaning for annotations?

Or shall we require `ComponentId` annotation and then have the name and description in a separate AgentDescription annotation?

Reply:
I think `@ComponentId` sounds nice for consistency, and I guess it sort of is the same concrete type of component as it already is for existing components

Reply:
+1 for `ComponentId`

Reply:
about `value`, the reason we use value is because you don't have to name the field. 

Like in `@ComponentId("foobar")` instead of `@ComponentId(name = "foobar")`

I think Spring has some tricks to deal with it, for example using a `@Alias` in the annotation field. 

Then, if the `value` is not defined, they look for values in the Alias. 
Anyway, if we find it important we can add similar facilities.

Reply:
It's now ComponentId and AgentDescription